### PR TITLE
Add bazel-cache chart

### PR DESCRIPTION
**Pull request by jpbower**
Bazel cache chart derived from docker image at [buchgr/bazel-remote](https://github.com/buchgr/bazel-remote).

### CODE REVIEW

File: bazel-cache/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.service.internalPort }}
+              protocol: TCP
+          volumeMounts:
+            - name: bazel-cache-root
+              subPath: data
+              mountPath: /data
```
Comment (Zetten) [inline bazel-cache/templates/deployment.yaml:51]: The container needs to be started with arguments to configure the bazel-remote service. At a minimum `--max_size=<PVC size>` but a generic value array and matching iterator here for custom startup args would be useful too.

**Review by Zetten — CHANGES_REQUESTED** [review]

**Comment by jpbower** [comment]
I made the requested changes and the chart seems to deploy fine. I have a strong feeling that I've misused the way you're meant to amend commits on GitHub but there we are...

File: bazel-cache/values.yaml [hunk 2]
```diff
@@ -19,6 +19,35 @@ service:
...
+# Path to a .htpasswd file. This flag is optional. Please read https://httpd.apache.org/docs/2.4/programs/htpasswd.html.
+#htpasswd_file: ""
+
+# Bool specifying whether or not to start the server with tls.  If true, server_cert and server_key flags are required.
+#tls_enabled: false
+
+# Path to a PEM encoded certificate file.  Required if tls_enabled is set to true.
+#tls_cert_file: ""
+
+#tls_key_file: ""
```
Comment (Zetten) [inline bazel-cache/values.yaml:43]: These values aren't used in the chart, but the fact they're here implies they could be. You could omit them and just leave `extraArgs`.

File: bazel-cache/values.yaml [hunk 3]
```diff
@@ -19,6 +19,35 @@ service:
...
+
+## Additional command line arguments to pass to nginx-ingress-controller
```
Comment (Zetten) [inline bazel-cache/values.yaml:45]: Our chart isn't nginx-ingress-controller

File: bazel-cache/values.yaml [hunk 4]
```diff
@@ -19,6 +19,35 @@ service:
...
+## E.g. to specify the default SSL certificate you can use
+## extraArgs:
+##   default-ssl-certificate: "<namespace>/<secret_name>"
+extraArgs: {}
```
Comment (Zetten) [inline bazel-cache/values.yaml:49]: Maybe add a comment linking to https://github.com/buchgr/bazel-remote/ so people can look up what args are available.

File: bazel-cache/templates/deployment.yaml [hunk 5]
```diff
@@ -26,6 +26,21 @@ spec:
...
+            - -port
+            - "{{ .Values.port }}"
+          {{- range $key, $value := .Values.extraArgs }}
+            {{- if $value }}
+            - -{{ $key }}
+            - "{{ $value }}"
+            {{- else }}
+            - -{{ $key }}
+            {{- end }}
+          {{- end }}
```
Comment (Zetten) [inline bazel-cache/templates/deployment.yaml:43]: I'm not sure this is the neatest or easiest way to do it. Maybe just expect an array of strings, not a dict, and do:
```
{{- if .Values.extraArgs }}
{{ toYaml .Values.extraArgs | indent 10 }}
{{- end }}
```
(from https://github.com/kubernetes/charts/blob/master/stable/kubernetes-dashboard/templates/deployment.yaml#L35 - check the indent as well)

**Review by Zetten — CHANGES_REQUESTED** [review]

**Pull request closed without merging**